iOS Client/TabController/AppCoordinator.swift

```diff
@@ -43,6 +44,51 @@ class AppCoordinator: NSObject, Coordinator, UITabBarControllerDelegate, StoreSu
             self.addChildCoordinator(childCoordinator: coord)
             return coord.rootViewController
         })
+        promptForNotifications()
+    }
+    
+    func promptForNotifications() {
+        let center = UNUserNotificationCenter.current()
+        
+        center.requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+            if granted {
+                MBStore().syncDevotions { devotions, error in
```

Comment:
New thought: How bout, in the DevotionCoordinator `newState`, we check to see if there are devotions. If there are, then we call promptForNotifications (and pass the devotions to it as a parameter), and then it simply asks the user for permission, and, if granted, calls scheduleNotifications on the devotions?

Reply:
and `newState` or `promptForNotifications` could be smart about checking if the user has already been prompted or not?